File: onyx-database/src/com/onyx/stream/QueryStream.java
package com.onyx.stream;

import com.onyx.persistence.manager.PersistenceManager;

import java.util.Objects;

/**
 * Created by tosborn1 on 5/19/16.
 *
 * This is a lambda for the Onyx Database Stream API.  This will iterate through entities
 */
@FunctionalInterface
public interface QueryStream<T>
{
    /**
     * Performs this operation on the given arguments.
     *
     * @param t the first input argument
     * @param u the second input argument is a PersistenceManager
     */
    void accept(T t, PersistenceManager u);
}

File: onyx-database/src/main/kotlin/com/onyx/persistence/stream/QueryStream.kt
package com.onyx.persistence.stream

import com.onyx.persistence.manager.PersistenceManager

/**
 * Created by Tim Osborn on 5/19/16.
 *
 * This is a lambda for the Onyx Database Stream API.  This will iterate through entities
 */
interface QueryStream<in T> {
    /**
     * Performs this operation on the given arguments.
     *
     * @param entity Managed entity
     * @param persistenceManager the second input argument is a PersistenceManager
     */
    fun accept(entity: T, persistenceManager: PersistenceManager)
}
